### DISCUSSION TITLE
Listners from LautFM @ Azuracast ?

### DISCUSSION THREAD

**Discussion by RadioHerzblut** [AzuraCast/AzuraCast#6738] [General]
**It goes here in the stream of LautFM .**

https://laut.fm

I have the current Azuracast Docker with version 0.19.2

The stream servers of LautFM are supplied by Azuracast ( AutoDJ and LIVE ).

There are instructions in the LautFM forum that you can read out the listeners in Azuracast. However, this does not work with the new version. 

Do you have any idea what I can do?

**Comment by BusterNeece** [comment]
@RadioHerzblut Can you link to those instructions on the forum?

**Reply by RadioHerzblut** [reply]
https://forum.laut.fm/t/azuracast-lautfm-einbindung/11671


The handset display should work with the "Rolling Relase". 

I activated this a few days ago. But there are no handset numbers to be seen there either.

I just wanted to switch to the rolling release version with Docker | Azuracast 0.19.2. After installing it, the Azuracast is no longer accessible. This problem is in another post.

**Comment by RadioHerzblut** [comment]
It's about the stream servers of LautFM in Germany

They are Icecast servers. 

A user reports that the listener numbers can be accessed via a relay with the following API URL:

https://MYSTATIONNAME.stream.laut.fm:8443/status-json.xsl?mount=/MYSTATIONNAME

Current rolling release until 18.11.2023 .  

I have installed the current version on my server until 18.11.2023. But without success of the listener numbers. 

Even the new version from brings no change.

Can you help here?

**Reply by LordHelmchen666** [reply]
MYSTATIONNAME is the laut.fm one, Not from AzuraCast. Just to be clear for BusterNeece

**Reply by RadioHerzblut** [reply]
Yes MYSTATIONNAME ist from LautFM

**Comment by LordHelmchen666** [comment]
Here is an example answer.
`{"icestats":{"source":{"bitrate":128,"genre":"Pop,Heavy Metal,Hard Rock","audio_info":"channels=2;samplerate=44100;bitrate=128","listeners":"3","channels":2,"samplerate":44100,"server_type":"audio\/mpeg","listenurl":"\/eisradio"}}}`
Is it different from what AzuraCast expects as answer?

**Comment by LordHelmchen666** [comment]
@BusterNeece 
This is how they explain we should fill in credentials in above tutorial.
![Screenshot_20231121_160046](https://github.com/AzuraCast/AzuraCast/assets/43477041/188c5992-5364-421a-93a6-ebc667e39a0d)
In the logs i get following entries
```
[2023-11-21T15:07:18.506221+00:00] AzuraCast.ERROR: Request returned status code 401. {"body":""} {"station":{"id":3,"name":"LautFM Eisradio"}}
[2023-11-21T15:07:18.509280+00:00] AzuraCast.ERROR: Request returned status code 404. {"body":""} {"station":{"id":3,"name":"LautFM Eisradio"}}
[2023-11-21T15:07:18.510912+00:00] AzuraCast.ERROR: Request returned status code 404. {"body":""} {"station":{"id":3,"name":"LautFM Eisradio"}}
[2023-11-21T15:07:18.517809+00:00] AzuraCast.ERROR: Request returned status code 404. {"body":""} {"station":{"id":3,"name":"LautFM Eisradio"}}
```
So i guess AzuraCast can't login. The login credentials should be
Username: source
Password: xxxxxxxx

So URL would be https://source:xxxxxx@MYSTATIONNAME.stream.laut.fm:8443/status-json.xsl?mount=/MYSTATIONNAME
How to reach this?

**Comment by RadioHerzblut** [comment]
@BusterNeece 
Do you see a possibility to solve the problem in a timely manner?
Would be very grateful for it ;-)

**Comment by RadioHerzblut** [comment]
Can you help here @ BusterNeece ?

**Reply by BusterNeece** [reply]
@RadioHerzblut No need to keep tagging me on this.

I'd basically need the laut.fm people to work with me so I can set up some sort of test environment where I can actually test things out, as this back-and-forth isn't doing me any favors as far as testing goes. Can you have one of them reach out to me so we can set something like that up?

**Reply by RadioHerzblut** [reply]
I will try that. Thank you very much for your answer.  I'll be in touch.

**Reply by RadioHerzblut** [reply]
Hello BusterNeece.  

I am happy to activate you as an editor for my LautFM stream.

I would send you the data and tester knowledge by e-mail. 
Is that ok for you?

So you can test.

**Reply by BusterNeece** [reply]
@RadioHerzblut That's fine.

**Reply by LordHelmchen666** [reply]
Buster if you need i can give you also access to my laut.fm account. Feel free to contact me.

**Comment by BusterNeece** [comment]
Looking into this with test credentials and I can tell you with some certainty that this is 100% an intentional decision on Laut.fm's part to obfuscate or limit the amount of data returned in their standardized Icecast stat points, and there may not be anything we can do about it on our end.

1) They require authentication for the otherwise public `/status-json.xsl` file, and then don't actually display the currently playing track in its API response.

2) All of their API responses in admin endpoints are very intentionally changed from `icestats` to `lautstats`, as in:

```xml
<lautstats credits="paeddy">
  <sources>
    <source>
      <mount>/radio</mount>
      <nodes>5</nodes>
      <status>1</status>
    </source>
  </sources>
</lautstats>
```

...and they also intentionally don't return the data that would be expected of those endpoints on stock Icecast.

Given the open-source nature of this project, if they're intentionally changing those stats endpoints to prevent us from being able to reliably pull listener details into software like AzuraCast, any change I make and any work I put into working around it would simply just result in them working around it some other way down the road.

I'm not inclined to spend any time in pursuit of supporting something that goes against standard behavior in a way that is both specific to this one organization and seemingly intentionally done.

If the folks at laut.fm care to reach out to me directly to clear this issue up and instruct me on how officially to proceed with adding support for their project into AzuraCast, they're welcome to e-mail me at any time. Until then, though, I'd expect this will continue to not work as expected.

**Reply by RadioHerzblut** [reply]
Thank you Bustrer for your detailed answer.

The following API query works without a user + password.
You can't get any further with this either ?

I get the "listener numbers" directly in the browser.

https://api.laut.fm/station/sprockytown/listeners

**Reply by BusterNeece** [reply]
Well, if what we were looking for was just a single scalar listener number, then yeah...but the problem is that our remote Now Playing adapters are expecting "better" responses than what we're getting from Laut.fm, like with track names and listener client details if we supply an admin password. I can understand how the latter is omitted on account of GDPR or what not, but right now we don't support _just_ returning a listener count from a remote relay. At least not elegantly.

**Reply by RadioHerzblut** [reply]
I tested with the people from Laut. The problem was recognised at Laut FM. A FIX should be coming soon.

**Comment by RadioHerzblut** [comment]
- The following URL displays the listener figures for LAUT FM on Azuracast:

http://stationsname.stream.laut.fm:8080?mount=/stationsname

or 

https://stationname.stream.laut.fm:8443?mount=/stationname


**At "Remote Sender Administrator Passwort "  your STATION Passwort.**

![image](https://github.com/AzuraCast/AzuraCast/assets/150465082/2fc7d877-8dbc-4e7e-9e8d-a7e14bc55722)

**Reply by BusterNeece** [reply]
@RadioHerzblut That's certainly something we can accommodate. In the latest Rolling Release, you won't need to put the "?mount=asdf" part for this to work as expected.

**Comment by RadioHerzblut** [comment]
**The connection now also works as follows. Thanks ;-)** 

![image](https://github.com/AzuraCast/AzuraCast/assets/150465082/40b3b3d3-9856-415e-a167-d207082d95b8)



If I connect as a LIVE DJ on port 8005, I can send LIVE . 

However, the listener numbers are not displayed in the broadcasting software ( mAirList ).

Is there a setting option here?

**Reply by 1Chaos71** [reply]
[GERMAN]
hast du da mittlerweile erfolg gehabt oder hab ich in diesen Beiträgen hier etwas überlesen?

[ENGLISH]
have you been successful in the meantime or have I missed something in these posts?

**Comment by Moonbase59** [comment]
Back in my time with mAirList (years ago), we had to "simulate" an Icecast XML response that returned the aggregate listener count of all actual streaming servers. Actually, mAirList was happy with a very minimal XML at that time, something like:

```xml
<?xml version="1.0"?>
<icestats>
  <source mount="/dummy">
    <listeners>233</listeners>
  </source>
</icestats>
```

So we used a little PHP script to return just that and set up a "dummy" streamer in mAirList with an option like (from memory, name might have changed) "Do not stream, only count listeners".

Since AzuraCast now seems able to aggregate listener counts, maybe you could write a little PHP that accesses the API, gets the real listener count, and return it as XML like above. Then point mAirList at that?

See also: https://community.mairlist.com/t/add-dummy-streaming-server-to-collect-of-listeners-from-external-source/13192/14

@BusterNeece: Would the nowplaying API listener count actually return the aggregated listener count of all AzuraRelays + laut.fm relays? Might eventually be important for GEMA/GVL (German  performance rights collecting society) reports.

**Reply by RadioHerzblut** [reply]
I have found a solution in mAirList.

You can create an encoder that only counts. 

![image](https://github.com/AzuraCast/AzuraCast/assets/150465082/4042a9da-7b07-40af-b160-3a30e9d88478)


Would of course be cool if Azuracast could ;-) 

But for now there is a solution.

**Reply by Moonbase59** [reply]
Yup, that’s exactly what I meant. But as-is, that will only return the laut.fm count. I seem to remember mAirList shows counts per stream, right (giving you several "Stream: count" listings)?

**Reply by RadioHerzblut** [reply]
A COUNTER encoder must be created for each query. 
I once made the suggestion to put this in a dialogue and that you can set up several queries.

I connect to Azuracast and send my programme to two Icecast streams.
